src/pg_query_parse_summary.c

```diff
@@ -0,0 +1,832 @@
+								String* relname = lsecond(list);
+								add_table(state, ContextDDL, schemaname->sval, relname->sval);
+							}
+						}
+					}
+					break;
+				}
+				case OBJECT_RULE:
+				case OBJECT_TRIGGER: {
+					ListCell* lc = NULL;
+					foreach(lc, stmt->objects) {
+						Node* obj = lfirst(lc);
+						if (obj->type == T_List) {
+							List* olist = (List*)obj;
+
+							// Ignore the last string (the rule/trigger name)
+							List* list = list_copy(olist);
+							list = list_truncate(list, list_length(list) - 1);
+
+							size_t len = list_length(list);
+							if (len == 0)
+								continue;
+
+							if (len == 1) {
+								String* relname = linitial(list);
+								add_table(state, ContextDDL, NULL, relname->sval);
+							} else if (len == 2) {
+								String* schemaname = linitial(list);
+								String* relname = lsecond(list);
+								add_table(state, ContextDDL, schemaname->sval, relname->sval);
+							}
+						}
+					}
+					break;
+				}
+
+				case OBJECT_FUNCTION: {
+					Node* obj_node = linitial(stmt->objects);
+
+					if (obj_node != NULL && obj_node->type == T_ObjectWithArgs) {
+						ObjectWithArgs* obj = (ObjectWithArgs*)obj_node;
+
+						if (obj->objname)
+							add_function(obj->objname, ContextDDL, state);
+					}
+
+					break;
+				}
+
+				default:
+					break;
+			}
+			return false; // Not handled by tree walker.
+		}
+
+		case T_GrantStmt: {
+			GrantStmt* stmt = (GrantStmt*)node;
+			List* objects = stmt->objects;
+
+			// Neither Ruby nor Rust handle OBJECT_COLUMN and OBJECT_SEQUENCE.
+			//
+			// Ruby just says "FIXME" for both, with no context provided.
+			// https://github.com/pganalyze/pg_query/blob/main/lib/pg_query/parse.rb
+			//
+			// Rust doesn't even acknowledge anything but OBJECT_TABLE.
+			if (stmt->objtype == OBJECT_TABLE) {
+				ListCell* lc = NULL;
+				foreach(lc, stmt->objects) {
+					add_from_clause_item(lfirst(lc), ContextDDL, state);
+				}
+			}
+
+			return false; // Not handled by tree walker.
+		}
+
+		case T_LockStmt: {
+			LockStmt* stmt = (LockStmt*)node;
+
+			ListCell* lc = NULL;
+			foreach(lc, stmt->relations) {
+				add_from_clause_item(lfirst(lc), ContextDDL, state);
+			}
+
+			return false; // Not handled by tree walker.
+		}
+
+		case T_ExplainStmt: {
+			ExplainStmt* stmt = (ExplainStmt*)node;
+			pg_query_parse_summary_walk_impl(stmt->query, state);
+			return false; // Not handled by tree walker.
+		}
+
+		case T_CreateFunctionStmt: {
+			CreateFunctionStmt* stmt = (CreateFunctionStmt*)node;
+			add_function(stmt->funcname, ContextDDL, state);
+			return false; // Not handled by tree walker.
+		}
+
+		case T_RenameStmt: {
+			RenameStmt* stmt = (RenameStmt*)node;
+			if (stmt->renameType == OBJECT_FUNCTION) {
+				ObjectWithArgs* obj = (ObjectWithArgs*)stmt->object;
+				// Add the original name.
+				add_function(obj->objname, ContextDDL, state);
+				// Add the new name.
+				add_function_cstr(stmt->newname, ContextDDL, state);
```

Comment:
```suggestion
				add_function(list_make1(makeString(stmt->newname)), ContextDDL, state);
```

(that way we don't need the extra `add_function_cstr` helper)